index.js

```diff
@@ -38,7 +41,18 @@ LBS.write = function (rec) {
   if (!rec) throw new Error('nothing passed to log');
   if (!this.writable) throw new Error('failed to write to a closed stream');
   if ('function' === typeof this.transform) rec = this.transform(rec);
-  this._logger.log(this._resolveLevel(rec.level), rec);
+  try {
+    this._logger.log(this._resolveLevel(rec.level), rec);
+  } catch (err) {
+    if (err instanceof _logentriesError.LogentriesError) {
+      console.error(err);
+      if (this.failOnError === undefined || this.failOnError) { process.exit(1); }
```

Comment:
I prefer `throw err` instead of `process.exit` as it retains existing behaviour and can allow other error handlers to catch.